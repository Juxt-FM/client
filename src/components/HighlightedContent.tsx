/**
 * @author Andrew Perera
 * Copyright (C) 2020 - All rights reserved
 */

import { useAuthStatus } from "../lib/context";

import Watchlists from "./Watchlists";

import styles from "../styles/modules/highlighted-content.module.scss";

interface ISection {
  title: string;
  children: JSX.Element;
}

const Section = ({ title, children }: ISection) => {
  return (
    <div className={styles.section}>
      <div className={styles.header}>
        <h4>{title}</h4>
      </div>
      {children}
    </div>
  );
};

const Trending = () => {
  const loggedIn = useAuthStatus();

  const renderLists = () => {
    return <div className={styles.empty}></div>;
  };

  return <Section title="Trending">{renderLists()}</Section>;
};

const Suggested = () => {
  const renderLists = () => {
    return <div className={styles.empty}></div>;
  };

  return <Section title="Suggested">{renderLists()}</Section>;
};

const HighlightedContent = () => {
  return (
    <div className={styles.root}>
      <Suggested />
      <Trending />
    </div>
  );
};

export default HighlightedContent;
