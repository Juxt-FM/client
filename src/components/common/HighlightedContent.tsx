/**
 * @author Andrew Perera
 * Copyright (C) 2020 - All rights reserved
 */

import { ReactChild, ReactChildren } from "react";

import styles from "../../styles/common/highlighted-content.module.scss";

interface ISection {
  title: string;
  children: JSX.Element;
}

export const HighlightedSection = ({ title, children }: ISection) => {
  return (
    <div className={styles.section}>
      <div className={styles.header}>
        <h4>{title}</h4>
      </div>
      {children}
    </div>
  );
};

interface IHighlightedContent {
  children: ReactChild | ReactChildren | ReactChild[] | ReactChildren[];
}

const HighlightedContent = (props: IHighlightedContent) => (
  <div className={styles.root}>{props.children}</div>
);

export default HighlightedContent;
