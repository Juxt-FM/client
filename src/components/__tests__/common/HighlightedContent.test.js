/**
 * @author Andrew Perera
 * Copyright (C) 2020 - All rights reserved
 */

import { create } from "react-test-renderer";

import HighlightedContent, {
  HighlightedSection,
} from "../../common/HighlightedContent";

describe("<HighlightedContent />", () => {
  it("should render children", () => {
    const component = create(
      <HighlightedContent>
        <div id="content">hello</div>
      </HighlightedContent>
    );

    const content = component.root.findByProps({ id: "content" });

    expect(content.children).toContain("hello");
  });
});

describe("<HighlightedSection />", () => {
  it("should render content appropriately", () => {
    const component = create(
      <HighlightedSection title="Test Title">
        <div id="content">hello</div>
      </HighlightedSection>
    );

    const tree = component.toJSON();

    const content = component.root.findByProps({ id: "content" });

    expect(content.children).toContain("hello");

    expect(tree).toMatchSnapshot();
  });
});
