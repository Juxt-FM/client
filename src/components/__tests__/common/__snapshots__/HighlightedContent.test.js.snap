// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HighlightedSection /> should render content appropriately 1`] = `
<div>
  <div>
    <h4>
      Test Title
    </h4>
  </div>
  <div
    id="content"
  >
    hello
  </div>
</div>
`;
